misc/test_bit_array_vectorization.py

```diff
@@ -0,0 +1,59 @@
+import taichi as ti
+
+ti.init(debug=True, cfg_optimization=False, kernel_profiler=True)
+
+vectorize = False
+
+ci1 = ti.type_factory_.get_custom_int_type(1, False)
+
+x = ti.field(dtype=ci1)
+y = ti.field(dtype=ci1)
+
+N = 4096
+block_size = 4
+bits = 32
+boundary_offset = 64
```

Comment:
Since each block has `1024x1024` pixels, maybe your boundary offset should be 1024? Note that the only sparse layer is the pointer layer, where you have `4x4` children, each can be active or not. So all your `NxN` pixels are active.

In this program, since you are accessing only `x[i, j]`, everything should be fine. However, when you proceed to `x[i, j - 1]`, there will be access out of bound.

Also, `n_blocks` seems a better name compared to `block_size`. The latter sounds like the number of pixels in a block.